src/pcre2_match.c

```diff
@@ -6119,7 +6122,11 @@ fprintf(stderr, "++ %2ld op=%3d %s\n", Fecode - mb->start_code, *Fecode,
       /* It is the end of whole-pattern recursion. */
 
       offset = Flast_group_offset;
-      if (offset == PCRE2_UNSET) return PCRE2_ERROR_INTERNAL;
+      if (offset == PCRE2_UNSET)
+        {
+        PCRE2_DEBUG_UNREACHABLE();
```

Comment:
maybe `PCRE2_ASSERT(0)`?

Reply:
As above, I would love to understand what the advantage of using `PCRE2_ASSERT(0)` might be.

Reply:
it is functionally equivalent but reduces the cognitive load of thinking about marking something UNREACHABLE when it "might" happen under some circumstances. It might be as flawed though and has the disadvantage of the failure (if it happens) not be that clear, so I dunno.

IMHO (like we saw in the autopossess case), once it getst better documentation around it we can come out with a much better overall solution and then instead of having an UNREACHABLE we end up with a clearer ASSERT.

Reply:
OK, let me look at this one another time...

Reply:
Essentially, we're searching a linked list of heapframes to try to find one which is of type GF_CAPTURE. If `offset == PCRE2_UNSET`, I think it means we reached the end of the linked list without finding it... just studying the code some more...

Reply:
If that `DEBUG_UNREACHABLE` is reached, I think it means the regex bytecode is malformed. Before OP_CLOSE, there should always be another op which creates a heapframe for the capturing group (such as OP_CBRA).

That sounds like a pretty reasonable place to put DEBUG_UNREACHABLE to me, but anything is fine.